exporters/otlp/src/otlp_grpc_client.cc

```diff
@@ -23,6 +31,73 @@ namespace exporter
 namespace otlp
 {
 
+#ifdef ENABLE_ASYNC_EXPORT
+
+namespace
+{
+struct OPENTELEMETRY_LOCAL_SYMBOL OtlpGrpcAsyncCallDataBase
+{
+  using GrpcAsyncCallback = bool (*)(OtlpGrpcAsyncCallDataBase *);
+
+  std::unique_ptr<google::protobuf::Arena> arena;
+  grpc::Status grpc_status;
+  std::unique_ptr<grpc::ClientContext> grpc_context;
+
+  opentelemetry::sdk::common::ExportResult export_result =
+      opentelemetry::sdk::common::ExportResult::kFailure;
+  GrpcAsyncCallback grpc_async_callback = nullptr;
+
+  OtlpGrpcAsyncCallDataBase() {}
+  virtual ~OtlpGrpcAsyncCallDataBase() {}
```

Comment:
A struct with a virtual ? Either use a class, or remove the virtual destructor.

Reply:
Thanks and one.